test/v3/runtime-core/apiLifecycle.spec.ts

```diff
@@ -0,0 +1,354 @@
+import { VueConstructor } from 'vue';
+const Vue = require('vue/dist/vue.common.js') as VueConstructor;
+import {
+  onBeforeMount,
+  onMounted,
+  ref,
+  createElement as h,
+  onBeforeUpdate,
+  onUpdated,
+  onBeforeUnmount,
+  onUnmounted,
+} from '../../../src';
+import { nextTick } from '../../helpers/utils';
+
+// reference: https://vue-composition-api-rfc.netlify.com/api.html#lifecycle-hooks
+
+describe('api: lifecycle hooks', () => {
+  it('onBeforeMount', () => {
+    const root = document.createElement('div');
+    const fn = jest.fn(() => {
+      // should be called before inner div is rendered
+      expect(root.innerHTML).toBe(``);
+    });
+
+    const Comp = {
+      setup() {
+        onBeforeMount(fn);
+        return () => h('div');
+      },
+    };
+    new Vue(Comp).$mount(root);
+    //render(h(Comp), root);
+    expect(fn).toHaveBeenCalledTimes(1);
+  });
+
+  it('onMounted', () => {
+    const root = document.createElement('div');
+    const fn = jest.fn(() => {
+      // should be called after inner div is rendered
+      expect(root.outerHTML).toBe(`<div></div>`);
+    });
+
+    const Comp = {
+      setup() {
+        onMounted(fn);
+        return () => h('div');
+      },
+    };
+    new Vue(Comp).$mount(root);
+    //render(h(Comp), root);
+    expect(fn).toHaveBeenCalledTimes(1);
+  });
+
+  it('onBeforeUpdate', async () => {
+    const count = ref(0);
+    // const root = document.createElement('div');
+    const fn = jest.fn(() => {
+      // should be called before inner div is updated
+      expect(vm.$el.outerHTML).toBe(`<div>0</div>`);
+    });
+
+    const Comp = {
+      setup() {
+        onBeforeUpdate(fn);
+        return () => h('div', (count.value as unknown) as string);
+      },
+    };
+    const vm = new Vue(Comp).$mount();
+    //render(h(Comp), root);
+
+    count.value = 1;
+    await nextTick();
+    expect(fn).toHaveBeenCalledTimes(1);
+  });
+
+  it('onUpdated', async () => {
+    const count = ref(0);
+    // const root = document.createElement('div');
+    const fn = jest.fn(() => {
+      // should be called after inner div is updated
+      expect(vm.$el.outerHTML).toBe(`<div>1</div>`);
+    });
+
+    const Comp = {
+      setup() {
+        onUpdated(fn);
+        return () => h('div', (count.value as unknown) as string);
+      },
+    };
+    const vm = new Vue(Comp).$mount();
+    //render(h(Comp), root);
+
+    count.value++;
+    await nextTick();
+    expect(fn).toHaveBeenCalledTimes(1);
+  });
+
+  // it('onBeforeUnmount', async () => {
```

Comment:
Comment this test because couldn't make this work, this was brought from `vue-next` not sure if we should fix this